spec/takeaway_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject(:takeaway) { described_class.new(menu) }
+  let(:menu) { double(:menu, dishes: { "pepperoni" => 7.50, "hawaiian" => 6.50, "vegetarian" => 5.50 }) }
```

Comment:
Again, you mentioned isolating tests as a difficulty in your review, but this is a really nice implementation.

Because you used dependency injection in your production code, you have the option of passing in this menu double when you instantiate the instance of the takeaway class you use in your tests.  This removes any dependency on a real instance of the Menu.